projects/ngx-piwik-pro/src/lib/services/cookie-management/cookie-management.service.ts

```diff
@@ -1,74 +1,43 @@
+import { CookieManagement } from '@piwikpro/tracking-base-library'
 import { Injectable } from '@angular/core';
-import { PaqService } from '../../services/paq/paq.service';
-import { TRACK_EVENT } from '../../constants/track-event.constant';
+
+type ICookieManagement = typeof CookieManagement;
 
 @Injectable({
   providedIn: 'root'
 })
 export class CookieManagementService {
-
-  constructor(
-    private readonly paqService: PaqService
-  ) {}
-
-  disableCookies() {
-    this.paqService.push([TRACK_EVENT.DISABLE_COOKIES]);
-
+  disableCookies(...params: Parameters<ICookieManagement['disableCookies']>) {
+    CookieManagement.disableCookies(...params);
   }
-
-  enableCookies() {
-    this.paqService.push([TRACK_EVENT.ENABLE_COOKIES]);
-
+  enableCookies(...params: Parameters<ICookieManagement['enableCookies']>) {
+    CookieManagement.enableCookies(...params);
   }
-
-  deleteCookies() {
-    this.paqService.push([TRACK_EVENT.DELETE_COOKIES]);
-
+  deleteCookies(...params: Parameters<ICookieManagement['deleteCookies']>) {
+    CookieManagement.deleteCookies(...params);
   }
-
-  hasCookies(): Promise<boolean> {
-    return new Promise((resolve, reject) => {
-      try {
-        this.paqService.push([
-          function (this: any) {
-            resolve(this.hasCookies());
-          },
-        ]);
-      } catch (e) {
-        if (e instanceof ReferenceError) {
-          reject(e);
-        }
-      }
-    });
+  hasCookies(...params: Parameters<ICookieManagement['hasCookies']>): Promise<boolean> {
```

Comment:
Suggestion: There are still few explicit return types defined. Would you please take another look and remove them all? You can search for ">): " in this diff to find them

Reply:
done